salt/states/zpool.py

```diff
@@ -30,6 +30,23 @@
               /dev/disk2
               /dev/disk3
 
+    partitionpool:
+      zpool.present:
+        - config:
+            import: false
+            force: true
+        - properties:
+            comment: disk partition salty storage pool
+            ashift: '12'
+            feature@lz4_compress: enabled
+        - filesystem_properties:
+            compression: lz4
+            atime: on
+            relatime: on
+        - layout:
+            disk-0:
```

Comment:
Right, sorry to keep bothering you about what is only an example.  I've managed to borrow a minion to test on and now I've found the errors that were occurring.

Using:
```
    - layout:
        - /dev/disk/by-uuid/12946d54-1c2f-44ca-b93b-44463422362b
```

Results in:

```
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1733, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1652, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/zpool.py", line 170, in present
                  layout[root_dev] = layout[root_dev].keys() if isinstance(layout[root_dev], OrderedDict) else layout[root_dev].split(' ')
              TypeError: list indices must be integers, not str
```

If I try:

```
    - layout: /dev/disk/by-uuid/12946d54-1c2f-44ca-b93b-44463422362b
```

I get:

```
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1733, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1652, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/zpool.py", line 170, in present
                  layout[root_dev] = layout[root_dev].keys() if isinstance(layout[root_dev], OrderedDict) else layout[root_dev].split(' ')
              TypeError: string indices must be integers, not str
```

But if I use the method given in the commit:

```
    - layout:
        disk-0:
          /dev/disk/by-uuid/12946d54-1c2f-44ca-b93b-44463422362b
```

It works without any errors.